projects/storefrontlib/src/shared/components/carousel/carousel.component.ts

```diff
@@ -1,78 +1,104 @@
 import {
+  ChangeDetectionStrategy,
   Component,
   ElementRef,
   EventEmitter,
   Input,
   OnInit,
   Output,
+  TemplateRef,
 } from '@angular/core';
 import { Observable } from 'rxjs';
 import { tap } from 'rxjs/operators';
 import { ICON_TYPE } from '../../../cms-components/misc/icon/index';
-import { CarouselItem } from './carousel.model';
 import { CarouselService } from './carousel.service';
 
+/**
+ * Generic carousel component that can be used to render any carousel items,
+ * such as products, images, banners, or any component. Carousel items are
+ * rendered in so-called carousel slides, and the previous/next buttons as well as
+ * the indicator-buttons can used to navigate the slides.
+ *
+ * The component uses an array of Observables (`items$`) as an input, to allow
+ * for lazy loading of items.
+ *
+ * The number of items per slide is calculated with the `itemWidth`, which can given
+ * in pixels or percentage.
+ *
+ * To allow for flexible rendering of items, the rendering is delegated to the
+ * given `template`. This allows for maximum flexibility.
+ */
 @Component({
   selector: 'cx-carousel',
   templateUrl: './carousel.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CarouselComponent implements OnInit {
+  /**
+   * The title is rendered as the carousel heading.
+   */
   @Input() title: string;
 
-  private _items: CarouselItem[];
-  @Input('items')
-  set items(value: CarouselItem[]) {
-    this._items = value;
-    this.select();
-  }
-  get items(): CarouselItem[] {
-    return this._items;
-  }
+  /**
+   * The items$ represent the carousel items. The items$ are
+   * observables so that the itesm can be loaded on demand.
+   */
+  @Input() items$: Observable<any>[];
```

Comment:
Would consider removing `$` from name, because it suggests that it's an observable, but actually it's an array (of observables), so it can be confusing.